builder/vsphere/supervisor/step_watch_source.go

```diff
@@ -92,7 +94,33 @@ func (s *StepWatchSource) Run(ctx context.Context, state multistep.StateBag) mul
 	return multistep.ActionContinue
 }
 
-func (s *StepWatchSource) Cleanup(state multistep.StateBag) {}
+func (s *StepWatchSource) Cleanup(state multistep.StateBag) {
+	if cleanImage, ok := state.GetOk(StateKeyCleanImportedImage); ok && cleanImage.(bool) {
+		logger := state.Get("logger").(*PackerLogger)
+
+		if state.Get(StateKeyImportedImageName) == nil {
+			logger.Info(fmt.Sprintf("Skipping clean up of imported image since config %s is not set.", StateKeyImportedImageName))
+			return
+		}
+		importImageName := state.Get(StateKeyImportedImageName).(string)
+
+		logger.Info(fmt.Sprintf("Deleting the imported ContentLibraryItem object %s in namespace %s.",
+			importImageName, s.Namespace))
+		ctx := context.Background()
+		importedImage := &imgregv1.ContentLibraryItem{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      importImageName,
+				Namespace: s.Namespace,
+			},
+		}
+		if err := s.KubeWatchClient.Delete(ctx, importedImage); err != nil {
+			logger.Error("error deleting the ContentLibraryItem object %s: %s", importImageName, err)
+		} else {
+			logger.Info(fmt.Sprintf("Successfully deleted the ContentLibraryItem object %s in namespace %s.",
+				importImageName, s.Namespace))
+		}
+	}
+}
```

Comment:
This can be done in the original `import_image` step, as the `Cleanup` functions are called in an reversed order (i.e. VM will be deleted before the imported image). See https://github.com/hashicorp/packer-plugin-sdk/blob/main/multistep/multistep.go#L52-L59 for more info.

Reply:
I probably missed how the cleanup works, if the source VM is not created successfully, will this image cleanup still be called (by putting in the import-image step? My original thought was to only clean the image if the source VM can be created and ready, so they can debug the issue by using the image if the source VM creation has issues.

Reply:
I wouldn't worry about the source VM status for cleaning up the imported image, since it's false by default. If users explicitly set it to true, they would expect it to get cleaned up after the builder completes.